src/Microsoft.CodeAnalysis.Analyzers/CSharp/MetaAnalyzers/CSharpDiagnosticAnalyzerAPIUsageAnalyzer.cs

```diff
@@ -14,6 +14,7 @@ protected override bool IsNamedTypeDeclarationBlock(SyntaxNode syntax)
             switch (syntax.Kind())
             {
                 case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.RecordDeclaration:
```

Comment:
Let's add a test for this.

Reply:
@Evangelink I'm not sure how the test should look like to have a different behavior before and after.

Reply:
I haven't tested but you might be able to copy a test raising an issue and change class to be record and show that this does raise an issue.

Reply:
@Evangelink tests are based on classes that inherit DiagnosticAnalyzer. Records on the other hand cannot inherit classes.
I'll debug this to make sure if records can come across this code path (e.g. record nested inside a class that inherit diagnostic analyzer)

Reply:
Oh I was left on an old proposition for the record where the inheritance was allowed but it seems it is no longer the case (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/records). So basically what's the purpose of your change of there is no way to reach this point?

Reply:
@Evangelink I didn't mean this is unreachable with my previous comment. (I also didn't mean this is reachable).

Until now I'm not sure whether this is reachable or not.
And if it's reachable, will this have a behavior change, or just returning early for some cases.

The method is called in two different places in the analyzer, which should have two different purposes. But the current implementation has struct, enum, interface. If all of these are reachable, then records shouldn't different.

I'll debug soon to get feel of how the code flows and get an understanding of it.